packages/grid/_modules_/grid/constants/eventsConstants.ts

```diff
@@ -1,85 +1,431 @@
 
+/**
+ * Fired when the selection state of one or multiple rows changes.
+ * Called with a [[GridSelectionModelChangeParams]] object.
+ * @event
+ */
 export const GRID_SELECTION_CHANGED = 'selectionChange';
 
+/**
+ * Fired when the current page has changed. Called with a [[GridPageChangeParams]] object.
+ * @event
+ */
 export const GRID_PAGE_CHANGED = 'pageChange';
+
+/**
+ * Fired when the page size has changed. Called with a [[GridPageChangeParams]] object.
+ * @event
+ */
 export const GRID_PAGESIZE_CHANGED = 'pageSizeChange';
 
-export const GRID_ROWS_SCROLL = 'scrolling:rows';
-export const GRID_ROWS_SCROLL_END = 'scroll:rowEnd';
+/**
+ * Fired during the scroll of the grid viewport. Called with a [[GridScrollParams]] object.
+ * @event
+ */
+export const GRID_ROWS_SCROLL = 'rowsScroll';
+
+/**
+ * Fired when scrolling to the bottom of the grid viewport. Called with a [[GridRowScrollEndParams]] object.
+ * @event
+ */
+export const GRID_ROWS_SCROLL_END = 'rowsScrollEnd';
 
+/**
+ * Fired when a `mousedown` DOM event happens in the column header separator.
+ * Called with a [[GridColumnHeaderParams]] object.
+ * @event
+ */
 export const GRID_COLUMN_SEPARATOR_MOUSE_DOWN = 'columnSeparatorMouseDown';
+
+/**
+ * Fired during the resizing of a column. Called with a [[GridColumnResizeParams]] object.
+ * @event
+ */
 export const GRID_COLUMN_RESIZE = 'columnResize';
+
+/**
+ * Fired when a column is resized. Called with a [[GridColumnResizeParams]] object.
+ * @event
+ */
 export const GRID_COLUMN_RESIZE_COMMITED = 'columnResizeCommited';
+
+/**
+ * Fired when the user starts resizing a column. Called with an object `{ field: string }`.
+ * @event
+ */
 export const GRID_COLUMN_RESIZE_START = 'columnResizeStart';
+
+/**
+ * Fired when the user starts resizing a column. Called with an object `{ field: string }`.
+ * @event
+ */
 export const GRID_COLUMN_RESIZE_STOP = 'columnResizeStop';
 
+/**
+ * Fired when the user ends resizing a column.
+ * @event
+ */
 export const GRID_COLUMN_ORDER_CHANGE = 'columnOrderChange';
 
+/**
+ * Fired when the rows are updated.
+ * @event
+ */
 export const GRID_ROWS_UPDATED = 'rowsUpdated';
+
+/**
+ * Fired when the rows are updated.
+ * @event
+ */
 export const GRID_ROWS_SET = 'rowsSet';
+
+/**
+ * Fired when the grid is emptied.
+ * @event
+ */
 export const GRID_ROWS_CLEARED = 'rowsCleared';
+
+/**
+ * Fired when the columns state is changed.
+ * Called with an array of strings correspoding to the field names.
+ * @event
+ */
 export const GRID_COLUMNS_UPDATED = 'columnsUpdated';
 
+/**
+ * Fired after the sort model is changed but before applying the new model.
+ * Called with a [[GridSortModelParams]] object.
+ * @event
+ */
 export const GRID_SORT_MODEL_CHANGE = 'sortModelChange';
+
+/**
+ * Fired after the filter model is changed and the new model is applied.
+ * Called with a [[GridFilterModelParams]] object.
+ * @event
+ */
 export const GRID_FILTER_MODEL_CHANGE = 'filterModelChange';
```

Comment:
Side note: the description of the `onFilterModelChange` prop is wrong. This event is published after applying the filter model, not before.

Reply:
It looks the implementation is wrong then. As far as I know, the application of the new model is a different concern it should always come after.

Regarding "Fired after the filter model is changed", IMHO "after" doesn't matter, it's about the user event action.

Reply:
Description updated.